types/block.go

```diff
@@ -10,7 +12,9 @@ import (
 
 	"github.com/gogo/protobuf/proto"
 	gogotypes "github.com/gogo/protobuf/types"
+	format "github.com/ipfs/go-ipld-format"
 
+	"github.com/lazyledger/lazyledger-core/p2p/ipld/plugin/nodes"
```

Comment:
From a dependencies perspective, it is probably desirable to not add any networking-related dependencies here to the types package. We can still merge this as is, but we should track this in an issue then. 

Note that the main reason to make this thing a method of `Block` was that we could potentially cache the extended data square as a field and reuse it for putting the block data to ipfs. Maybe we can achieve the same without adding the actual networking logic here. Either way, we can figure this out later.

Reply:
This a great point! You're right,`PutBlock` definitely does not have to be a method of `Block`. #196